addons/website_sale/models/product_product.py

```diff
@@ -422,35 +422,3 @@ def _prepare_gmc_additional_info(self):
             gmc_info['item_group_id'] = self.product_tmpl_id.id
 
         return gmc_info
-    # ---------------------------------------------------------
-    # Product Ribbon Methods
-    # ---------------------------------------------------------
-
-    def _get_ribbon(self, products_prices):
-        """
-        Retrieve a mapping from product id to it's assigned ribbon.
-
-        This method first checks if a ribbon is manually set If no manual ribbon is set, it returns
-        appropriate ribbon based on the priority assigned in product.ribbon.
-
-        :param products_prices: mapping from product's id to it's corresponding price in
-            shop.
-        :rtype: dict.
-        """
-        ProductRibbon = self.env['product.ribbon']
-        product_ribbon_map = {}
-        auto_assign_ribbons = ProductRibbon.search([('assign', '!=', 'manual')])
-        for product in self:
-            product_ribbon_map[product.id] = ProductRibbon
-            manually_set_ribbon = (
-                product.variant_ribbon_id or product.product_tmpl_id.website_ribbon_id
-            )
-
-            if manually_set_ribbon:
-                product_ribbon_map[product.id] = manually_set_ribbon
-            else:
-                product_ribbon_map[product.id] = auto_assign_ribbons._get_applicable_ribbon(
-                    product, products_prices.get(product.product_tmpl_id) if products_prices else {},
-                )
-
-        return product_ribbon_map
```

Comment:
@dere-odoo We moved this method to `product_template` because it's used on the `/shop` and `/product` pages, where we show templates. We added a variant argument to allow using specific variants in some cases, but otherwise it will automatically default to the first variant.